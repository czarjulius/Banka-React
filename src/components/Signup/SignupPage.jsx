import React from 'react';
import { Link } from 'react-router-dom';
import Navbar from '../Navbar/Navbar';
import Footer from '../Footer/Footer';
import '../../index.scss';
import '../Signup/signup.scss';

const SignupPage = () => (
  <div>
    <Navbar />
    <div className="register-container">
        
        <div className="register-form">
          <form>
              <div className="form-header">
                  <h4>Register </h4>
                  <p id="registerMessage"></p>
                  <p id="registerSuccess"></p>
                  <img src="./image/spinner.gif" id="spinner" />
              </div>
            <div className="input"> 
              <label for="firstname"> First name </label>
              <input type="text" id="firstname" required /> 
              <p id="firstnameError" className="alert-class"></p>
            </div>
            <div className="input"> 
              <label for="lastname"> Last name </label>
              <input type="text" id="lastname" required /> 
              <p id="lastnameError" className="alert-class"></p>
            </div>
            <div className="input"> 
              <label for="email"> Email </label>
              <input type="email" id="email" required /> 
              <p id="emailError" className="alert-class"></p>
            </div>
            <div className="input"> 
                <label for="phone"> Mobile No. </label>
                <input type="text" id="phone" required /> 
                <p id="phoneError" className="alert-class"></p>
              </div>
            <div className="input"> 
              <label for="password" > Password </label>
              <input type="password" id="password" required />
              <p id="passwordError" className="alert-class"></p>
            </div>
            <div className="input"> 
              <label for="confirmpassword" >Confirm Password </label>
              <input type="password" id="confirmpassword" required />
              <p id="confirmpassword-alert" className="alert-class"></p>
            </div>
              <p className="sign-up"> 
                    <input type="button" id="registerUser" value="Register" className="btn" />            
                </p>
          </form>
          <div className="form-footer">
            <span id="no_account">Already have an account ?</span>
            <a href="login.html"> Login </a>
          </div>
        </div>
      </div>
  <Footer />
  </div>
);

export default SignupPage;