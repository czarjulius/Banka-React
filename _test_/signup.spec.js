import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { SignupPage, mapDispatchToProps } from '../src/components/Signup/SignupPage';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const renderSignInPage = args => {
  const defaultProps = {
    isLoading: false,
    history: {},
    match: {},
    articles: {},
    newUser: jest.fn(),
    errors: undefined,
    location: {}
  };
  const props = { ...defaultProps, ...args };
  return mount(
    <BrowserRouter>
      <SignupPage {...props} />
    </BrowserRouter>
  );
};

describe('Sign up Components Tests', () => {
  it('renders Sign up template', () => {
    const wrapper = renderSignInPage();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
  });

  it(`Simulates an onchange event on form email input`, () => {
    const wrapper = renderSignInPage();
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { currentTarget: { value: 'fola@gmail.com' } });
  });

  it(`Simulates an onchange event on form password input`, () => {
    const wrapper = renderSignInPage();
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { currentTarget: { value: 'password' } });
  });
  it('Simulates a form submit event', () => {
    const wrapper = renderSignInPage();
    wrapper.find('form').simulate('submit');
  });

  it('should dispatch login request action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).newUser();
  });
});