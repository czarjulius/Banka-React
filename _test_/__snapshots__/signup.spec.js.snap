// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Components Tests renders Sign up template 1`] = `ReactWrapper {}`;
